### PR TITLE
ceph upgrade to v13.3.2: port WindRiver patches

### PR DESCRIPTION
**Pull request by changchengx**
Hi @dtroyer & @oponcea ,
  We've ported all the WR patches to stx-ceph/v13.2.2.
  1. Split the original big patches into small ones.
  2. Remove the WR patch list info in the commit message.
  3. Add some commit message in the generated small patches which are splited from big ones.
  4. ceph cluster works in one StarlingX by doing dedicated-storage deployment with below guide document in virtual environment:
       https://docs.starlingx.io/installation_guide/dedicated_storage.html#dedicated-storage
       Note: doesn't add cinder/glance/swift/nova service. 2 OSDs run-up in cluster.
 5. To check the original patches which these patches are ported from, please check below document
  https://etherpad.openstack.org/p/stx-ceph_upgrade

**Storage ID**

- https://storyboard.openstack.org/#!/story/2003605

**Task ID**

- 28857

Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>
Signed-off-by: Dehao Shang <dehao.shang@intel.com>

### CODE REVIEW

**Comment by iotcg** [comment]
@dtroyer 
If you need to keep the WindRiver patch list info in the commit message, please tell us the format requirement.

**Comment by changchengx** [comment]
@dtroyer @oponcea 
Could you help review this pull request. I have verified it works.

File: stx/mgr-restful-plugin [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    rst=$?
+    if [ $rst -eq 0 ]; then
+        echo "ceph mgr restful plugin is enabled, need swith port to $PORT"
+    else
+        echo "ceph mgr restful plugin can't be enabled, abort"
+        return
+    fi
+
+    #ceph config-key set mgr/restful/${user_name}/server_port $PORT
+    ceph config-key set mgr/restful/server_port $PORT
```
Comment (yhu6) [inline stx/mgr-restful-plugin]: "set" might not work with ceph v13.2.2,  need to use "put"

File: stx/mgr-restful-plugin [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    ceph restful create-self-signed-cert
```
Comment (yhu6) [inline stx/mgr-restful-plugin]: in ~/restful/module.py, "create-self-signed-cert" actually includes a "restart", so you don't need to run the following "ceph restful restart" again.
Instead, to save one more "restart", you could execute "ceph restful create-key admin" prior to "ceph restful create-self-signed-cert", and then make a comment to tell why another "ceph restful restart" is not needed explicitly.

File: stx/mgr-restful-plugin [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    ceph restful create-key admin
+#    ceph mgr module enable restful
+    ceph restful restart
+    restful_check
```
Comment (yhu6) [inline stx/mgr-restful-plugin]: After "restful" is restarted either by "ceph restful create-self-signed-cert" or "ceph restful restart", we cannot assure "restful_check" will read out results within expected period of time.
So, you might have to reconsider the implementation of  "restful_check"!!

File: stx/ceph-rest-api [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Ceph REST API daemon
+# Description:       Ceph REST API daemon
+### END INIT INFO
+
+DESC="ceph-rest-api"
+DAEMON="/usr/bin/ceph-rest-api"
+RUNDIR="/var/run/ceph"
+PIDFILE="${RUNDIR}/ceph-rest-api.pid"
```
Comment (yhu6) [inline stx/ceph-rest-api]: this ceps-rest-api.pid path is NOT consistent with the path you specified in "service-mgmt/sm-db-1.0.0/database/create_sm_db.sql"
see https://review.openstack.org/#/c/634120/ in details.
Either way you need to modify one place or another.

Comment (changchengx) [reply stx/ceph-rest-api]: @yhu6
1) Currently, the system won't execute ceph-rest-api even ceph-rest-api file are built into system 
2) I'm going to delete stx/{ceph-rest-api,ceph-rest-api.service} file from source code.
3) stx/mgr-restful-plugin has set the PID file as /var/run/ceph/mgr/mgr-restful-plugin.pid

Have you hit error in the running system?

File: stx/mgr-restful-plugin (hunk 2 above)
Comment (changchengx) [reply stx/mgr-restful-plugin]: @yhu6 Thanks Yong.
restart will be auto triggered in the end of "create-self-signed-cert".
I'll refine it.

File: stx/mgr-restful-plugin (hunk 1 above)
Comment (changchengx) [reply stx/mgr-restful-plugin]: “put” is deprecated from luminous(ver12) and ceph prefer to using “set” to do some configuration.
     What’s the problem hit in your side when using “put”?

     I’ve checked the source code: vim cgcs-root/stx/git/ceph/src/mon/ConfigKeyService.cc +215
     “put” and “set” are in the same implementation. The patch is: https://github.com/ceph/ceph/commit/4eb1a518e3ed52103218616cd

Comment (changchengx) [reply stx/mgr-restful-plugin]: @yhu6 
I'll use below command to remove the warning:
ceph config set mgr mgr/restful/server_port $PORT

Comment (yhu6) [reply stx/mgr-restful-plugin]: I am good with it!

File: stx/ceph-rest-api (hunk 4 above)
Comment (yhu6) [reply stx/ceph-rest-api]: just check this path (/var/run/ceph/mgr/mgr-restful-plugin.pid) to make sure it is the same path in sm db file.

**Pull request closed without merging**